### ISSUE TITLE
Integrated Terminal environment setting doesn't work

### ISSUE THREAD

**Issue by tjx666** [microsoft/vscode#123433]
```
Version: 1.57.0-insider (Universal)
Commit: a19565da7dfc739fa26e36aae96113e19fe3806f
Date: 2021-05-07T18:11:54.098Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Darwin x64 20.4.0
```

reproduce:

1. paste follow setting:
```json
{
    "terminal.integrated.env.osx": {
        "YTJ_FLUTTER_VERSION": "latest"
    },
}
```
2. open new terminal and run follow command
```bash
echo $YTJ_FLUTTER_VERSION 
```

![image](https://user-images.githubusercontent.com/41773861/117603576-13bfc100-b186-11eb-863e-f0fd22e6f94f.png)

**Comment by nitishxp** [comment]
Set `"terminal.integrated.allowWorkspaceConfiguration": true in settings.json

`

**Comment by nitishxp** [comment]
Enable Allow workspace configuration 

![image](https://user-images.githubusercontent.com/25002490/117616314-921d6280-b188-11eb-8229-da81fec30f95.png)

**Comment by tjx666** [comment]
@nitishxp It works, but I think there should be a notification to user when workspace exists terminal settings.

**Comment by tjx666** [comment]
And should also be note in setting description:

![image](https://user-images.githubusercontent.com/41773861/117625861-1089fc80-b1a9-11eb-99ec-0e45128988cb.png)

**Comment by Paladiamors** [comment]
I agree with this comment. I wasted hours trying to figure out the problem and online vscode documentation was unhelpful as I also tried to set the environment variable from a `.env` file which did not work. 

I did not run into this problem in previous versions of vs code for OSX as VSCode would ask to allow modifications and set this. This is currently no longer being done.

> Enable Allow workspace configuration
> 
> ![image](https://user-images.githubusercontent.com/25002490/117616314-921d6280-b188-11eb-8229-da81fec30f95.png)
